addons/point_of_sale/models/pos_order.py

```diff
@@ -960,24 +960,25 @@ def set_pack_operation_lot(self, picking=None):
                             if stock_production_lot.product_id.tracking == 'lot':
                                 qty = abs(pos_pack_lot.pos_order_line_id.qty)
                             qty_done += qty
-                            pack_lots.append({'lot_id': stock_production_lot.id, 'qty': qty})
+                            pack_lots.append({'lot': stock_production_lot, 'qty': qty})
```

Comment:
I'would have done the stuff with the quant location here, keep the key lot_id and add a key location_id